lib/devise/strategies/two_factor_pam_authenticatable.rb

```diff
@@ -22,7 +22,7 @@ def authenticate!
       protected
 
       def valid_params?
-        params[scope] && params[scope][:password].present?
+        params[scope].respond_to?(:[]) && params[scope][:password].present?
```

Comment:
Would

```ruby
params.dig(scope, :password).present?
```

work as well? That is a bit more succinct and imo even a bit more readable.

Reply:
No! This would precisely not work, because the failing test passes an array as `scope`, which causes `no implicit conversion of Symbol into Integer`, an error that would remain the same with `dig`!

Reply:
Ah, OK. Thanks for clarifying!